nemo/core/classes/modelPT.py

```diff
@@ -1350,31 +1120,16 @@ def cfg(self, cfg):
         self._set_hparams(self._cfg)
         self._hparams_initial = copy.deepcopy(self._hparams)
 
-    @staticmethod
-    def _make_nemo_file_from_folder(filename, source_dir):
-        with tarfile.open(filename, "w:gz") as tar:
-            tar.add(source_dir, arcname=".")
-
-    @staticmethod
-    def _unpack_nemo_file(path2file: str, out_folder: str) -> str:
-        if not path.exists(path2file):
-            raise FileNotFoundError(f"{path2file} does not exist")
-        tar = tarfile.open(path2file, "r:gz")
-        tar.extractall(path=out_folder)
-        tar.close()
-        return out_folder
-
     @staticmethod
     def _is_model_being_restored() -> bool:
-        global _MODEL_IS_RESTORED
-        return _MODEL_IS_RESTORED
+        app_state = AppState()
+        return app_state.model_is_restored
```

Comment:
Just a note: if I understand it correctly, the new solution has the same flaw as the old one, i.e. it won't properly handle cases where more than 1 model object is instantiated

Reply:
@titu1994 was able to add multi-model instantiation so I think it works.

Reply:
Ah missed this comment. Multimodal restoration works now that apostate manages model IDs at a global level. Any number of Nemo models can be instantiated - either via restore or from constructor